sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -274,16 +281,79 @@ protected override async ValueTask<Response> UpdateStatusAsync(bool async, Cance
             }
         }
 
-        private static Response GetResponseFromState(OperationState<T> state)
+        private static Response GetResponseFromState(OperationState<T> state, RequestMethod? requestmethod = null)
         {
             if (state.HasSucceeded)
             {
                 return state.RawResponse;
             }
 
+            // if this is a fake delete lro with 404, just return empty response with 200
+            if (RequestMethod.Delete == requestmethod && state.RawResponse.Status == 404)
+            {
+                return new EmptyResponse(HttpStatusCode.OK);
```

Comment:
Are we confident we're transferring all the data an end-user will need to the fake response type?  We know that no one will check this for content/headers, etc?

Reply:
For fake delete, we are sure the response is empty, maybe it makes more sense to change the response status to 204? as Michael suggested [here](https://github.com/Azure/azure-sdk-for-net/pull/41088#discussion_r1504672038)

Reply:
updated to 204